File: lsd-interceptors/src/test/java/io/lsdconsulting/interceptors/aop/SpringDataRepositoryInterceptorTest.java
package io.lsdconsulting.interceptors.aop;

import org.junit.jupiter.api.Test;

import static org.mockito.Mockito.mock;

class SpringDataRepositoryInterceptorTest {

    private final AopInterceptorDelegate delegate = mock(AopInterceptorDelegate.class);
    private final SpringDataRepositoryInterceptor interceptor = new SpringDataRepositoryInterceptor(delegate);

    @Test
    void trapExceptionsToPreventBreakingBuilds() {
        interceptor.captureRepositoryResponses(null, null);
        interceptor.captureRepositoryErrors(null, null);
    }
}

File: lsd-interceptors/src/test/kotlin/io/lsdconsulting/interceptors/aop/SpringDataRepositoryInterceptorTest.kt
package io.lsdconsulting.interceptors.aop

import io.mockk.mockk
import org.aspectj.lang.JoinPoint
import org.junit.jupiter.api.Test

internal class SpringDataRepositoryInterceptorTest {
    private val delegate = mockk<AopInterceptorDelegate>()
    private val interceptor = SpringDataRepositoryInterceptor(delegate)

    @Test
    fun trapExceptionsToPreventBreakingBuilds() {
        interceptor.captureRepositoryResponses(mockk<JoinPoint>(), null)
        interceptor.captureRepositoryErrors(mockk<JoinPoint>(), Exception())
    }
}
